src/proxy/http/profiles.rs

```diff
@@ -16,7 +20,13 @@ use never::Never;
 use super::retry::Budget;
 use NameAddr;
 
-pub type Routes = Vec<(RequestMatch, Route)>;
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ConcreteDst(pub NameAddr);
+
+pub struct Routes {
```

Comment:
This name feels slightly inappropriate now, since it's no longer just a list of routes...